### ISSUE TITLE
creating end to end marine-waste detection

### ISSUE THREAD

**Issue by NIKITA320495** [abhisheks008/ML-Crate#610]
### ML-Crate Repository (Proposing new issue)
:red_circle: **Project Title** :End-to end marine waste detection
:red_circle: **Aim** :to detect marine waste 
:red_circle: **Dataset** :marine-dabris dataset
:red_circle: **Approach** : End to end marine waste detection model using yolo and html,css and js trained on sonar images

**Comment by abhisheks008** [comment]
Please share a detailed approach for this project.

**Comment by NIKITA320495** [comment]
@abhisheks008 
This project focuses on marine waste detection using forward-looking sonar cameras on ships, leveraging deep learning and computer vision. By training a YOLO model on the marine-debris dataset of underwater images and integrating it with a Flask-based frontend, it  ensures real-time detection and timely intervention of marine waste and hazards, enhancing maritime safety and environmental protection across diverse water conditions.

**Comment by abhisheks008** [comment]
What are the models you are planning for this?
@NIKITA320495

**Comment by NIKITA320495** [comment]
@abhisheks008  yolo8n.pt pretrained model

**Comment by abhisheks008** [comment]
Assigned you the churn prediction project

@NIKITA320495

**Comment by adi271001** [comment]
@abhisheks008 can you please assign this to me will use 3-4models

**Comment by abhisheks008** [comment]
> @abhisheks008 can you please assign this to me will use 3-4models

Share your approach with the name of the models you are planning to implement?
Mention 6-7 models.

**Comment by adi271001** [comment]
> > @abhisheks008 can you please assign this to me will use 3-4models
> 
> Share your approach with the name of the models you are planning to implement? Mention 6-7 models.

my approach is data prepocessing , extracting features  via methods like histograms , shape feature , texture feature using gabor filters , keypoint descriptors using sift algorithm and finally training the model using ml modles like 1. LOgistic regression 2. gradient boost 3. xgboost 4.random forest 5. svm 6. knn 7. decision trees etc and test them by dividing the dataste and plotting accuracy graphs and correlation heatmaps.

**Comment by abhisheks008** [comment]
> > > @abhisheks008 can you please assign this to me will use 3-4models
> > 
> > 
> > Share your approach with the name of the models you are planning to implement? Mention 6-7 models.
> 
> my approach is data prepocessing , extracting features via methods like histograms , shape feature , texture feature using gabor filters , keypoint descriptors using sift algorithm and finally training the model using ml modles like 1. LOgistic regression 2. gradient boost 3. xgboost 4.random forest 5. svm 6. knn 7. decision trees etc and test them by dividing the dataste and plotting accuracy graphs and correlation heatmaps. please assign it to me under ssoc 2024

Waiting for a confirmation from @NIKITA320495 as she opened this issue for her own contributions.

**Comment by NIKITA320495** [comment]
@abhisheks008 sir, if there's any chance that @adi271001 and I can colaborate and then the issue is assigned to both of us , i will be glad to do so otherwise i want to work on this on my own after the completion of my bank churn prediction project

**Comment by abhisheks008** [comment]
> @abhisheks008 sir, if there's any chance that @adi271001 and I can colaborate and then the issue is assigned to both of us , i will be glad to do so otherwise i want to work on this on my own after the completion of my bank churn prediction project

Hi @adi271001 and @NIKITA320495 as per the code of conduct of this repo, one issue can be assigned to one contributor only. As it is a competiton, we can't divide the points among the contributors, that's why we can't collab you guys together.

@NIKITA320495 you can work on this issue after the completion of the assigned one. As you have opened this issue for your own contribution, you have the right to lock it for your own.

@adi271001 you can check out other open issues, opened by me and having Up for Grabs tag.

**Comment by NIKITA320495** [comment]
hi @abhisheks008 can you please assign me this issue??

**Comment by abhisheks008** [comment]
Assigned to you @NIKITA320495
